### PR TITLE
Register service and build number with discovery

### PR DESCRIPTION
**Pull request by jason-qq**
Checks for BuildNumber Attribute in DiscoveryService metadata and either includes/excludes BuildNumber from post body in Add-DiscoveryRegistration

### CODE REVIEW

File: common/Fabric-Install-Utilities.psm1 [hunk 1]
```diff
@@ -585,6 +585,55 @@ function Write-Console($message){
     Write-Host $message -ForegroundColor Gray
 }
 
+function Test-DiscoveryHasBuildVersion($discoveryUrl, $credential) {
+    $response = [xml](Invoke-WebRequest -Method Get -Uri "$discoveryUrl/`$metadata" -Credential $credential -ContentType "application/xml")
+	
+	return $response.Edmx.DataServices.Schema.EntityType.Property.Name -contains 'BuildNumber'
+}
+
+function Add-DiscoveryRegistration($discoveryUrl, $serviceUrl, $credential, $buildVersion, $serviceName, $serviceVersion, $friendlyName, $description) {
```
Comment (jason-qq) [inline common/Fabric-Install-Utilities.psm1]: The Add-DiscoveryRegistration function exists in both the Identity/Authorization install scripts, bringing it here brings in a few more parameters to the function ($serviceName, $serviceVersion, $friendlyName, $description)

**Review by bsmithvt — APPROVED** [review]

File: common/Fabric-Install-Utilities.psm1 (hunk 1 above)
Comment (ryanorbaker) [reply common/Fabric-Install-Utilities.psm1]: I'm assuming we are already passing the build version in during existing calls?

**Review by ryanorbaker — APPROVED** [review]

File: common/Fabric-Install-Utilities.psm1 (hunk 1 above)
Comment (michaelvidal24) [reply common/Fabric-Install-Utilities.psm1]: Can we encapsulate those parameters (including $buildVerson) into an object? That would reduce the number of parameters in the function signature, perhaps making it more readable.

File: common/Fabric-Install-Utilities.psm1 [hunk 2]
```diff
@@ -646,4 +695,6 @@ Export-ModuleMember -Function Add-ServiceUserToDiscovery
 Export-ModuleMember -Function Invoke-Sql
 Export-ModuleMember -Function Read-FabricInstallerSecret
 Export-ModuleMember -Function Get-ErrorFromResponse
-Export-ModuleMember -Function Invoke-ResetFabricInstallerSecret
\ No newline at end of file
+Export-ModuleMember -Function Invoke-ResetFabricInstallerSecret
+Export-ModuleMember -Function Test-DiscoveryHasBuildVersion
```
Comment (michaelvidal24) [inline common/Fabric-Install-Utilities.psm1]: Do we need to export the `Test-DiscoveryHasBuildVersion` function? Seems like right now it is only used internally in the module.

File: common/Fabric-Install-Utilities.psm1 (hunk 1 above)
Comment (jason-qq) [reply common/Fabric-Install-Utilities.psm1]: Yes, the specific install scripts know about which directory they are in making it easier to find their dll's.  
Will put $buildVersion, $serviceName, $serviceVersion, $friendlyName, $description into an object, I think it also makes sense to bring in $serviceUrl as well

File: common/Fabric-Install-Utilities.psm1 [hunk 3]
```diff
@@ -585,6 +585,55 @@ function Write-Console($message){
...
+function Test-DiscoveryHasBuildVersion($discoveryUrl, $credential) {
+    $response = [xml](Invoke-WebRequest -Method Get -Uri "$discoveryUrl/`$metadata" -Credential $credential -ContentType "application/xml")
+	
+    return $response.Edmx.DataServices.Schema.EntityType.Property.Name -contains 'BuildNumber'
+}
+
+function Add-DiscoveryRegistration($discoveryUrl, $credential, $discoveryPostBody) {
+    $hasVersion = Test-DiscoveryHasBuildVersion $discoveryUrl $credential
+
+    if($hasVersion) {
```
Comment (michaelvidal24) [inline common/Fabric-Install-Utilities.psm1]: You could reduce the duplicated code here by first setting all the common properties on the registration body. Then if the $hasVersion variable is true, then have a statement to set `$registrationBody.BuildNumber = $discoveryPostBody.buildVersion`.

Comment (jason-qq) [reply common/Fabric-Install-Utilities.psm1]: Oh neat, will do that

**Review by michaelvidal24 — APPROVED** [review]

**Pull request merged**